File: botlist/tests/misc/ruby/TestGroupLinks.rb
##
## Berlin Brown
## TestSections.rb
##
## Also see:
## http://www.ruby-doc.org/stdlib/libdoc/test/unit/rdoc/

require 'java'

include_class "junit.framework.Test"
include_class "junit.framework.TestCase"
include_class "junit.framework.TestSuite"
include_class 'junit.textui.TestRunner'

include_class "org.spirit.test.JRubyTestCase"

class TestGroupLinks < JRubyTestCase
  
  attr_accessor :ac
  def initialize(context)
    @ac = context
  end
  def test()
    c = @ac.getBean("radController")
    dao = c.linkGroupsDao
    n = "http://www.kjslfs22222.com"
    linkGroup = dao.readLinkGroup("aaf9dfb546f650d5fa614156000info")
    linkGroupId = linkGroup.id
    puts linkGroup.groupName
  end
  
  def to_s
    "Current Test"
  end
end

t = TestGroupLinks.new($context)
t.test()
t

## End of Script

File: botlist/tests/misc/py/test_group_links.py
##
## Berlin Brown
##
import random
import all_tests
import junit
from java.lang import Long

from org.spirit.bean.impl import BotListGroupLinks

class testGroupLinks (junit.framework.TestCase):
    
    def __init__(self, name):
        junit.framework.TestCase.__init__(self, name)
        self.theTestFunction = getattr(self, name)
        
    def setUp(self):
        pass
    
    def tearDown(self):
        pass
    
    def runTest(self):
        self.theTestFunction()
        
    def testAddNewGroupLink(self):
        randlinkid = random.randint(1, 90000)
        newrandlink = "%s%s.com" % ("http://www.new", randlinkid)
        
        c = applicationContext.getBean("radController")
        dao = c.linkGroupsDao
        daoGroupLinks = c.groupLinksDao
        
        self.assertNotNull(dao)        
        linkGroup = dao.readLinkGroup("aaf9dfb546f650d5fa614156000info")
        self.assertNotNull(linkGroup)

        linkGroupId = linkGroup.id
        sessionFactory = dao.getSessionFactory()
        hbm_session = sessionFactory.openSession()
        tx = hbm_session.beginTransaction()
        linkGroupBean = hbm_session.load("org.spirit.bean.impl.BotListLinkGroups", linkGroupId, None)
        self.assertNotNull(linkGroupBean)
        # Error with jython proxying, unable to save one-to-many
        
        link = BotListGroupLinks()        
        link.groupId = 4
        link.urlTitle = 'mytitle'
        link.mainUrl = newrandlink
        link.keywords = 'abc'

        daoGroupLinks.createGroupLink(link)
        tx.commit()
        hbm_session.close()
        
    def testGroupLinks(self):
        c = applicationContext.getBean("radController")
        dao = c.linkGroupsDao
        self.assertNotNull(dao)

        linkGroup = dao.readLinkGroup("aaf9dfb546f650d5fa614156000info")
        self.assertNotNull(linkGroup)
        self.assertEquals(linkGroup.generatedId, "aaf9dfb546f650d5fa614156000info")
        
def suite():
    testClassObject = testGroupLinks

    # Donot edit below.
    suite = junit.framework.TestSuite()
    # Edit the two class names
    for i in dir(testClassObject):
        if i.startswith("test"):
            p = testClassObject(i)
            suite.addTest(p)
            
    return suite

# End of Script
